src/SW_Domain.c

```diff
@@ -0,0 +1,153 @@
+
+    MyFileName = InFiles[eDomain];
+	f = OpenFile(MyFileName, "r", LogInfo);
+
+    // Set SW_DOMAIN
+    while(GetALine(f, inbuf)) {
+        sscanf(inbuf, "%s %s", key, value);
+
+        keyID = domain_inkey_to_id(key);
+        switch(keyID) {
+            case 0: // Domain type
+                if(strcmp(value, "xy") != 0 && strcmp(value, "s") != 0) {
+                    LogError(LogInfo, LOGFATAL, "%s: Incorrect domain type %s."\
+                             " Please select from \"xy\" and \"s\".",
+                             MyFileName, value);
+                }
+                strcpy(SW_Domain->DomainType, value);
+                break;
+            case 1: // Number of X slots
+                SW_Domain->nDimX = atoi(value);
+                break;
+            case 2: // Number of Y slots
+                SW_Domain->nDimY = atoi(value);
+                break;
+            case 3: // Number of S slots
+                SW_Domain->nDimS = atoi(value);
+                break;
+            case 4: // Start year
+                y = atoi(value);
+
+                if (y < 0) {
+                    CloseFile(&f, LogInfo);
+                    LogError(LogInfo, LOGFATAL,
+                             "%s: Negative start year (%d)", MyFileName, y);
+                }
+                SW_Domain->startyr = yearto4digit((TimeInt) y);
+                SW_Model->addtl_yr = 0; // Could be done anywhere; SOILWAT2 runs don't need a delta year
+                break;
+            case 5: // End year
+                y = atoi(value);
+
+                if (y < 0) {
+                    CloseFile(&f, LogInfo);
+                    LogError(LogInfo, LOGFATAL,
+                             "%s: Negative ending year (%d)", MyFileName, y);
+                }
+                SW_Domain->endyr = yearto4digit((TimeInt) y);
+                break;
+            case 6: // Start day of year
+                SW_Domain->startstart = atoi(value);
+                fstartdy = swTRUE;
+                break;
+            case 7: // End day of year
+                tempdoy = atoi(value);
+                SW_Domain->endend = (tempdoy == 365) ?
+                                Time_get_lastdoy_y(SW_Domain->endyr) : 365;
+                fenddy = swTRUE;
+                break;
+            case KEY_NOT_FOUND: // Unknown key
+                LogError(LogInfo, LOGWARN, "%s: Ignoring an unknown key, %s",
+                         MyFileName, key);
+                break;
+        }
+    }
+
+	if (SW_Domain->endyr < SW_Domain->startyr) {
+		CloseFile(&f, LogInfo);
+        LogError(LogInfo, LOGFATAL, "%s: Start Year > End Year", MyFileName);
+	}
+
+    // Check if start day of year was not found
+    if(!fstartdy) {
+        LogError(LogInfo, LOGWARN, "Domain.in: Missing Start Day - using 1\n");
+        SW_Domain->startstart = 1;
+    }
+
+	SW_Domain->startstart += ((SW_Model->isnorth) ? DAYFIRST_NORTH : DAYFIRST_SOUTH) - 1;
```

Comment:
Values of SW_DOMAIN cannot be a function of SW_MODEL,
```
SW_Domain->startstart += ((SW_Model->isnorth) ?  ...
```

The domain may cover both southern and northern hemisphere sites, but SW_Model is one specific site either in the southern or northern hemisphere.

Leave `SW_Domain->startstart` and `SW_Domain->endend` as inputed from "domain.in" or at most adjusted by other values from `SW_Domain`, e.g., `Time_get_lastdoy_y(SW_Domain->endyr)`